src/Makefile.am

```diff
@@ -8,7 +8,9 @@
 SUFFIXES = .cpp .h .yy .ll .cc .hh .hpp
 
 bin_PROGRAMS = souffle souffle-mcpp souffle-profile
- 
+
+BUILT_SOURCES = parser.hh stack.hh
```

Comment:
According to https://www.gnu.org/software/automake/manual/html_node/Sources.html BUILT_SOURCES is a giant hack that only works on the targets: `all`, `check`, and `install`. Any reason for not simply adding ParserDriver.h and scanner.ll to the dependency list in line 185 (and merge the list with the other dependency list in line 192)?